nimare/workflows/cbma.py

```diff
@@ -0,0 +1,177 @@
+        if obj == FWECorrector:
+            kwargs["method"] = obj_str
+
+    return _check_type(obj, clss, **kwargs)
+
+
+def cbma_workflow(
+    dataset,
+    estimator=None,
+    corrector=None,
+    diagnostics=None,
+    output_dir=None,
+    n_cores=1,
+):
+    """Compose a coordinate-based meta-analysis workflow.
+
+    .. versionadded:: 0.0.14
+
+    This workflow performs a coordinate-based meta-analysis, multiple comparison correction,
+    and diagnostics analyses on corrected meta-analytic maps.
+
+    Parameters
+    ----------
+    dataset : :obj:`~nimare.dataset.Dataset`
+        Dataset for which to run meta-analyses to generate maps.
+    estimator : :class:`~nimare.base.CBMAEstimator`, :obj:`str` {'ale', 'scale', 'mkdadensity', \
+    'kda'}, or optional
+        Meta-analysis estimator. Default is :class:`~nimare.meta.cbma.ale.ALE`.
+    corrector : :class:`~nimare.correct.Corrector`, :obj:`str` {'montecarlo', 'fdr', \
+    'bonferroni'} or optional
+        Meta-analysis corrector. Default is :class:`~nimare.correct.FWECorrector`.
+    diagnostics : :obj:`list` of :class:`~nimare.diagnostics.Diagnostics`, \
+    :class:`~nimare.diagnostics.Diagnostics`, :obj:`str` {'jackknife', 'focusCounter'}, \
+    or optional
+        List of meta-analysis diagnostic classes. A single diagnostic class can also be passed.
+        Default is :class:`~nimare.diagnostics.FocusCounter`.
+    output_dir : :obj:`str`, optional
+        Output directory in which to save results. If the directory doesn't
+        exist, it will be created. Default is None (the results are not saved).
+    n_cores : :obj:`int`, optional
+        Number of cores to use for parallelization.
+        If <=0, defaults to using all available cores.
+        If estimator, corrector, or diagnostics are passed as initialized objects, this parameter
+        will be ignored.
+        Default is 1.
+
+    Returns
+    -------
+    :obj:`~nimare.results.MetaResult`
+        Results of Estimator and Corrector fitting with cluster and diagnostic tables.
+    """
+    n_cores = _check_ncores(n_cores)
+
+    if not isinstance(diagnostics, list) and diagnostics is not None:
+        diagnostics = [diagnostics]
+
+    # Check dataset type
+    dataset = _check_type(dataset, Dataset)
+
+    # Options allows for string input
+    estm_options = ("ale", "scale", "mkdadensity", "kda")
+    corr_options = ("montecarlo", "fdr", "bonferroni")
+    diag_options = ("jackknife", "focuscounter")
+
+    # Check inputs and set defaults if input is None
+    estimator = (
+        ALE(n_cores=n_cores)
+        if estimator is None
+        else _check_input(estimator, CBMAEstimator, estm_options, n_cores=n_cores)
+    )
+    corrector = (
+        FWECorrector(method="montecarlo", n_cores=n_cores)
+        if corrector is None
+        else _check_input(corrector, Corrector, corr_options, n_cores=n_cores)
+    )
+
+    pval = 0.05 if corrector.method == "montecarlo" else 0.001
```

Comment:
Let's have 0.05 and 0.001 be defined as global variables defined at the top of the file that tells what they mean.

0.05 could be `CORRECTED_P_VALUE_THRESHOLD`
0.001 could be `UNCORRECTED_P_VALUE_THRESHOLD`